### PR TITLE
combine index-file-saving and params-saving into one API

### PR DESCRIPTION
**Pull request by eddogola**
## Description

bring together `_save_index` and `_save_params` into one function `save_checkpoint`. The entire module is used to write the index file, only in rank 0, while submodule parameters are saved to a file in each rank. The associated ckpt test is updated to reflect this change.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Feature/Issue validation/testing

- [x] Test LocalIndexMetadataTest

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
+docs/build
```
Comment (wz337) [inline .gitignore:1]: Let's remove this from the commit.

File: test/local_test_ckpt_index_file.py [hunk 2]
```diff
@@ -6,10 +6,7 @@
 import json
 import os
 
-from pippy.hf._SaveModule import (
-    _save_index,
-    _save_checkpoint,
-)
+from pippy.hf._SaveModule import save_checkpoint
```
Comment (wz337) [inline test/local_test_ckpt_index_file.py:9]: Also, I think we can update this test name to reflect what we are testing. 
local_test_save_load_checkpoint.py

File: test/local_test_ckpt_index_file.py [hunk 3]
```diff
@@ -94,10 +91,13 @@ def run_worker(args: List[str | int]) -> None:
     else:
         stage()
 
+    # Take an optimization step
+    optimizer.step()
+    ref = deepcopy(stage.submod.state_dict())
+    save_checkpoint(stage, stage.submod, CKPT_DIR)
```
Comment (fegin) [inline test/local_test_ckpt_index_file.py:97]: What's the reason to move the three lines here? Does this mean the original test case have an issue?

Comment (eddogola) [reply test/local_test_ckpt_index_file.py]: save_checkpoint writes the index as well as saving weights. As they were decoupled before, I was writing the index in the last set of lines and the checkpoints further up. Now, I had to bring save_checkpoint up to do the whole thing.

**Pull request merged**